src/main/java/com/google/cloud/tools/DeployAction.java

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.tools;
+
+import com.google.cloud.tools.ProcessCaller.Tool;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Deploys an app to the server.
+ */
+public class DeployAction extends Action {
+
+  private static Option[] acceptedFlags = {
+      Option.BUCKET,
+      Option.DOCKER_BUILD,
+      Option.FORCE,
+      Option.IMAGE_URL,
+      Option.PROMOTE,
+      Option.SERVER
+  };
+  private AppType appType;
+  private Action stageAction;
```

Comment:
Yes, as I mentioned earlier, StageAction is in DeployAction's constructor because it makes unit testing possible. Unless I add setStageAction() to inject a mock StageAction, which is uglier.

It is weird that the API offers a stage functionality that isn't documented anywhere in our public docs and that normal users don't know about. But since we're already having to inject StageActions into DeployActions, it's a small leap to having them execute staging on their own. I'll decouple.